### ISSUE TITLE
CTC loss appears to use raw logits instead of log-probabilities

### ISSUE THREAD

**Issue by KijitoraFinch** [gwx314/STARS#9]
Hi, thank you for releasing code and weights!

I'm currently fine-tuning STARS on Japanese data, and during training, I noticed negative `ph_ctc_loss`.

The training code seems to passes `ph_frame_logits` (raw logits) directly to `torch.nn.functional.ctc_loss` :
[tasks/stars/task.py#L521-L527](https://github.com/gwx314/STARS/blob/main/tasks/stars/task.py#L521-L527)

However, `torch.nn.functional.ctc_loss` expects log-probabilities as its first argument:
https://docs.pytorch.org/docs/2.4/generated/torch.nn.functional.ctc_loss.html

Without the normalization, the objective does not seem to be calculated correctly. (e.g. simply adding the same positive constant to all logits decreases the loss even though the corresponding softmax probabilities remain unchanged)

**Comment by gwx314** [comment]
Thank you so much for pointing this out and for the detailed explanation!
You are completely right—`torch.nn.functional.ctc_loss` expects log-probabilities. It's been a while since I ran those initial training experiments, and I'm not entirely sure if I had patched this locally on the renting server during training without synced back when uploading the codebase.
Thanks again for catching this! I have updated and fixed the code accordingly.
